hooks/useRequest.ts

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import { BASE_URL } from "../utils/constants";
+
+export const useRequest = <T>({
+  url,
+  method,
+}: {
+  url: string;
+  method: "GET" | "POST" | "PATCH" | "DELETE";
+}) => {
+  const [data, setData] = useState<T>({} as T);
+
+  const requestFetch = (body?: Record<string, any>) => {
+    let apiUrl = url;
+    if (method === "GET") {
+      const query = body
+        ? Object.keys(body).map((e) => {
+            return e + "=" + body[e];
+          })
+        : "";
+      apiUrl = url + "?" + query;
+    }
+    fetch(BASE_URL + apiUrl, {
+      method: method,
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: method === "GET" ? null : JSON.stringify(body),
+    })
+      .then((response) => {
+        if (response.status === 200) {
```

Comment:
ok 값으로 체크해주는게 좋을 것 같아요!
```suggestion
        if (response.ok) {
```